File: NavigationComponentsJava/app/src/main/java/com/muneiah/navigationcomponentsjava/HomeFragment.java
package com.muneiah.navigationcomponentsjava;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;


/**
 * A simple {@link Fragment} subclass.
 */
public class HomeFragment extends Fragment {

    public HomeFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view= inflater.inflate(R.layout.fragment_home, container, false);
        Button red=view.findViewById(R.id.redbtn);
        Button blue=view.findViewById(R.id.bluebtn);
        Button green=view.findViewById(R.id.greenbtn);
        red.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final NavController navController = Navigation.findNavController(v);
                //navController.popBackStack();
                navController.navigate(R.id.action_homeFragment_to_redFragment);
            }
        });
        blue.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final NavController navController = Navigation.findNavController(v);
                //navController.popBackStack();
                navController.navigate(R.id.action_homeFragment_to_blueFragment);
            }
        });
        green.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final NavController navController = Navigation.findNavController(v);
                //navController.popBackStack();
                navController.navigate(R.id.action_homeFragment_to_greenFragment);
            }
        });
        return view;
    }
}

File: MyNavComponentDemo/app/src/main/java/com/muneiah/mynavcomponentdemo/HomeFragment.kt
package com.muneiah.mynavcomponentdemo

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.navigation.findNavController

/**
 * A simple [Fragment] subclass.
 */
class HomeFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view=inflater.inflate(R.layout.fragment_home,container,false)
            val  ob=view.findViewById<Button>(R.id.frst)
            val  tb=view.findViewById<Button>(R.id.scnd)
            val  ttb=view.findViewById<Button>(R.id.trd)
        ob.setOnClickListener {
            it.findNavController().navigate(R.id.action_homeFragment_to_oneFragment)
        }
        tb.setOnClickListener {
            it.findNavController().navigate(R.id.action_homeFragment_to_twoFragment)
        }
        ttb.setOnClickListener {
            it.findNavController().navigate(R.id.action_homeFragment_to_threeFragment)
        }
        return view
    }



}
